### PR TITLE
[Guide] DB Root Credential Rotation

### PR DESCRIPTION
**Pull request by yhyakuna**
The first draft of the ***DB Root Credential Rotation*** to fulfill the [Asana task](https://app.asana.com/0/112957393038545/632193521989355/f). 

- Let me know if we need to add more ***story*** under the **Challenge** and **Solution** section
- Please review the technical accuracy
- Let me know if additional steps/parameters should be added to highlight the benefit of this feature

Also, please review the messaging:
- [Recommendation](https://github.com/hashicorp/vault/pull/4508/files#diff-d958bd2bee21ef2f27766f817bc4c752R58) to rotate the root credential immediately after the DB configuration was added
- A [brief warning](https://github.com/hashicorp/vault/pull/4508/files#diff-d958bd2bee21ef2f27766f817bc4c752R359) suggesting to create a dedicated superuser for Vault usages

This guide will look like --> https://drive.google.com/open?id=11MDeCbR6b87Ax4xt4zN2ooFzMN5WaejR

### CODE REVIEW

File: website/source/guides/secret-mgmt/db-root-rotation.html.md [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+
+This guide requires that you have a PostgreSQL server to connect to. If you
+don't have one, install [PostgreSQL](https://www.postgresql.org/download/) to
+perform the steps described in this guide.
+
+**Example on CentOS:**
+
+```shell
+# To list available packages
+$ yum list postgreql*
```
Comment (chrishoffman) [inline website/source/guides/secret-mgmt/db-root-rotation.html.md]: I'm not sure we want to provide install instructions for postgres here.  Maybe we can link to a procedure from the official documents.

File: website/source/guides/secret-mgmt/db-root-rotation.html.md [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+<br>
+
+This confirms that the Vault successfully connected to your PostgreSQL server
+and created a new user based on the privilege defined by `readonly.sql`.
+
+> The user credentials generated by the Vault has a limited TTL based on your
+configuration (`default_ttl`). In addition, you can revoke them if you suspect
+that it was compromised.
+
+**NOTE:** The `pg_hba.conf` is configured to prompt for password to
```
Comment (chrishoffman) [inline website/source/guides/secret-mgmt/db-root-rotation.html.md]: I think it is ok to show the psql commands above to verify the config but advanced configuration like this doesn't feel right in our guides.

File: website/source/guides/secret-mgmt/db-root-rotation.html.md [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+password. This is the same for all root database credentials given to Vault.
+Therefore, you should create a separate superuser dedicated to the Vault usage
+and not used for other purposes.  
+
+
+## Next steps
+
+In this guide, you learned how to rotate the root database credentials.
+
+Read the [Java Sample App using Spring Cloud
```
Comment (chrishoffman) [inline website/source/guides/secret-mgmt/db-root-rotation.html.md]: I'm not sure how we are doing the next steps in the guides sections but this doesn't seem related.

Comment (yhyakuna) [reply website/source/guides/secret-mgmt/db-root-rotation.html.md]: I couldn't think of the next step. :(   The Java app guide leverages both database secret engine and EaaS features of Vault, I linked it as a working example of the DB secret engine.  But you're right that it has nothing to do with DB root credential rotation.  I'll think of a better one to link.

File: website/source/guides/secret-mgmt/db-root-rotation.html.md (hunk 2 above)
Comment (yhyakuna) [reply website/source/guides/secret-mgmt/db-root-rotation.html.md]: I debated over this myself, so I'll remove it.   BTW, is there a better way to test/verify the feature?  I used the `psql` to test because I didn't know any other way.  But if you have a more elegant way to show that it worked, let me know.

File: website/source/guides/secret-mgmt/db-root-rotation.html.md (hunk 1 above)
Comment (yhyakuna) [reply website/source/guides/secret-mgmt/db-root-rotation.html.md]: Easy fix!  I'll do that. :)

**Comment by yhyakuna** [comment]
@chrishoffman   Thank you for reminding me!  I believe the [Asana task](https://app.asana.com/0/112957393038545/632193521989355/f) also mentioned that we should recommend rotating the credential after the initial configuration.  I'll add that!
